portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,45 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import com.google.gson.Gson;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<String> comments = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+        String json = convertToJson(comments);
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+      String comment = getComment(request);
+      comments.add(comment);
+      response.sendRedirect("/index.html");
+    }
+
+
+  private String convertToJson(ArrayList comments){
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
+    return json;
+  }
+
+    private String getComment(HttpServletRequest request) {
+        String comment = request.getParameter("comment-box");
```

Comment:
You can do this in 1 line. 

return request.getParameter("comment-box")

Reply:
Ive made this change :)